Makefile

```diff
@@ -94,19 +98,37 @@ export INCLUDE	:=	$(foreach dir,$(INCLUDES),-I$(CURDIR)/$(dir)) \
 export LIBPATHS	:=	-L$(LIBOGC_LIB) $(foreach dir,$(LIBDIRS),-L$(dir)/lib)
 
 export OUTPUT	:=	$(CURDIR)/$(TARGET)
-.PHONY: $(BUILD) clean
+.PHONY: all dol gci qoob qoobsx viper xeno compressed_dol compressed_gci $(BUILD) clean run
+
+export BUILD_MAKE := @mkdir -p $(BUILD) && $(MAKE) --no-print-directory -C $(BUILD) -f $(CURDIR)/Makefile
 
 #---------------------------------------------------------------------------------
-$(BUILD):
-	@[ -d $@ ] || mkdir -p $@
-	@$(MAKE) --no-print-directory -C $(BUILD) -f $(CURDIR)/Makefile
+all: dol gci qoob viper compressed_dol compressed_gci
+dol:
+	$(BUILD_MAKE) $(OUTPUT).dol
+gci:
+	$(BUILD_MAKE) $(OUTPUT).gci
+qoob:
+	$(BUILD_MAKE) $(OUTPUT).gcb
+qoobsx:
+	$(BUILD_MAKE) $(OUTPUT_SX).elf
+viper:
+	$(BUILD_MAKE) $(OUTPUT).vgc
+xeno:
+	$(BUILD_MAKE) $(OUTPUT)_xeno.dol
+compressed_dol:
+	$(BUILD_MAKE) $(OUTPUT)_xz.dol
+compressed_gci:
+	$(BUILD_MAKE) $(OUTPUT)_xz.gci
+
+#---------------------------------------------------------------------------------
+$(BUILD): all
 
 #---------------------------------------------------------------------------------
 clean:
 	@echo clean ...
-	@rm -fr $(BUILD) $(OUTPUT).{elf,dol} $(OUTPUT).{gcb,vgc} \
-		$(OUTPUT)_xz.{dol,elf,qbsx} $(OUTPUT_SX).{dol,elf} \
-		$(OUTPUT)_xeno.{bin,elf} $(OUTPUT){,_xz}.gci
+	@rm -rf $(BUILD)
+	@rm -f $(OUTPUT)* $(OUTPUT_SX)*
```

Comment:
Are you sure this is necessary? I was under the impression that make itself would expand this. Either way, I don't like blindly applying wildcards like this. We should retain the explicit file list, and change it to be shell-agnostic if needed.

Reply:
Yea the curly glob patterns don't work from make for me. Make nor the shell expands them. Not exactly sure why as I use `zsh` and those patterns work outside of make. I think it has something to do with make defaulting to using `sh` shell and/or extended globing being enabled: https://stackoverflow.com/a/12838683/2544290

I did a quick test:
```Makefile
test:
	@$$SHELL --version
	ls iplboot.{dol,elf}
```
```zsh
make test
# zsh 5.8 (x86_64-ubuntu-linux-gnu)
# ls iplboot.{dol,elf}
# ls: cannot access 'iplboot.{dol,elf}': No such file or directory
ls iplboot.{dol,elf}
# iplboot.dol  iplboot.elf
```

I see your point about unsafe wildcards. Ideally all build artifacts would be put under a folder that could be easily cleaned and ignored. I considered changing `OUTPUT` to `$(CURDIR)/build/$(TARGET)` so everything goes into `build` dir, but I didn't want to break anything downstream that expects the current output file locations. Thoughts?

Reply:
It might be that make uses sh. On Arch Linux and NixOS, that's bash, which supports this; but Debian for example has its own implementation (dash) as /bin/sh.
I don't think we have any downstreams that rely on output paths currently (pretty sure PicoBoot simply relies on copying the file by hand), so go ahead.

Reply:
Wilco. Moving build artifacts to the `build` folder and will update the `clean` target to just delete this folder.